docs/design.md

```diff
@@ -38,13 +38,112 @@ spec:
 ```
 
 ### Command
+
 Optionally you can override the default command of a container by providing
 `command` in the Substratus resource:
 ```yaml
 spec:
   command: ["train.sh"]
 ```
 
+### Storage
+
+#### Bucket Paths
+
+In Kubernetes the combination of the following fields make up a unique reference to a given object (i.e. database key): `split(.apiVersion, "/")[0] + .kind + .metadata.namespace + .metadata.name`. By storing related artifacts using a similar scheme, restore operations are made trivial: the administrator will not need to worry about backing up the `.status.url` field (as opposed to a scheme where storage location is `.metadata.uid` based - which would be unique for a given in-time instance of an object). When deleting and recreating clusters, if the storage bucket persisted, objects (i.e. Models, Datasets) can simply be re-applied into the new cluster (i.e. [velero](https://velero.io/)/similar is not needed). This plays nicely with GitOps.
+
+Storage scheme:
+
+```
+# Models
+gs://{project_id}-substratus-storage/namespaces/{namespace}/models/{name}/model
+gs://{project_id}-substratus-storage/namespaces/{namespace}/models/{name}/logs
+
+# Datasets
+gs://{project_id}-substratus-storage/namespaces/{namespace}/datasets/{name}/data
+gs://{project_id}-substratus-storage/namespaces/{namespace}/datasets/{name}/logs
+```
+
+NOTE: With this approach, a single bucket could be used to store all Models and Datasets for a given cluster.
```

Comment:
I'm trying to come up with a reason why the separation would be a good thing. Regulation for models has obviously not landed in a meaningful way anywhere but for customer data (and by extension, datasets), GDPR and friends are a consideration. Even within that regulatory scheme, we'd probably be fine mashing everything together.

I think I overall hedge toward keeping the bucket separation just because the uniform bucket access and IAM isolation makes permissioning specific SAs a little cleaner. Just my take!

Reply:
I think we will need to go granular with the permissions over time… eek, so that’s why I think single might not matter